wicket-core/src/test/java/org/apache/wicket/markup/html/form/AutoLabelLabelProviderLocatorTest.java

```diff
@@ -0,0 +1,154 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.wicket.markup.html.form;
+
+import java.util.ArrayList;
+import java.util.List;
+import org.apache.wicket.AttributeModifier;
+import org.apache.wicket.Component;
+import org.apache.wicket.markup.html.WebMarkupContainer;
+import org.apache.wicket.markup.html.WebPage;
+import org.apache.wicket.markup.html.basic.Label;
+import org.apache.wicket.markup.html.panel.Panel;
+import org.apache.wicket.markup.repeater.RepeatingView;
+import org.apache.wicket.model.IModel;
+import org.apache.wicket.model.Model;
+import org.apache.wicket.util.tester.WicketTestCase;
+import org.junit.jupiter.api.Test;
+
+/**
+ * Tests {@code wicket:for} attribute functionality using {@link ILabelProviderLocator}
+ */
+@SuppressWarnings({ "rawtypes", "serial" })
+class AutoLabelLabelProviderLocatorTest extends WicketTestCase
+{
+	static class ILabelProviderLocatorPanel1 extends Panel implements ILabelProviderLocator
+	{
+
+		private TextField<String> text;
+
+		ILabelProviderLocatorPanel1(String id)
+		{
+			super(id);
+			WebMarkupContainer dummy = new WebMarkupContainer("dummy");
+			dummy.add(AttributeModifier.replace("class", "dummy1"));
+			add(dummy);
+			text = new TextField<>("text", Model.of(""));
+			dummy.add(text);
+		}
+
+		@Override
+		public Component getAutoLabelComponent()
+		{
+			return text;
+		}
+	}
+
+	static class ILabelProviderLocatorPanel2 extends Panel implements ILabelProviderLocator
+	{
+
+		private TextField<String> text;
+
+		ILabelProviderLocatorPanel2(String id)
+		{
+			super(id);
+			WebMarkupContainer dummy = new WebMarkupContainer("dummy");
+			dummy.add(AttributeModifier.replace("class", "dummy"));
+			add(dummy);
+
+			WebMarkupContainer dummy1 = new WebMarkupContainer("dummy1");
+			dummy1.add(AttributeModifier.replace("class", "dummy1"));
+			dummy.add(dummy1);
+
+			text = new TextField<>("text", Model.of(""));
+			dummy1.add(text);
+		}
+
+		@Override
+		public Component getAutoLabelComponent()
+		{
+			return text;
+		}
+	}
+
+	private interface IEditPanelProvider
+	{
+		IModel<String> getLabelText();
+
+		Panel createEditPanel(String id);
+	}
+
+	static class EditPage extends WebPage
+	{
+		EditPage(List<IEditPanelProvider> editPanelProviders)
+		{
+			Form<Void> form = new Form<>("form");
+			add(form);
+			RepeatingView editRow = new RepeatingView("edit-row");
+			form.add(editRow);
+			for (IEditPanelProvider panelProvider: editPanelProviders) {
+				WebMarkupContainer mc = new WebMarkupContainer(editRow.newChildId());
+				mc.add(new Label("label", panelProvider.getLabelText()));
+				mc.add(panelProvider.createEditPanel("edit-component"));
+				editRow.add(mc);
+			}
+		}
+	}
+
+	@Test
+	void markupIntoLabelInsertion()
```

Comment:
The method name is not very clear what this test is supposed to do.

Reply:
Arg... copy paste from other test on same package.. renamed to testILabelProviderLocator